File: ml-pipeline/utils/load_prompt.js
import { readFileSync } from 'fs';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

const __dirname = dirname(fileURLToPath(import.meta.url));
const PROMPTS_DIR = join(__dirname, '..', 'prompts');

export function loadPrompt(filename, vars = {}) {
  const filePath = join(PROMPTS_DIR, filename);
  let template = readFileSync(filePath, 'utf-8');
  for (const [key, value] of Object.entries(vars)) {
    template = template.replaceAll(`{{${key}}}`, value);
  }
  return template;
}

File: ml_backend/backend/ml_pipeline/utils/load_prompt.py
from pathlib import Path

PROMPTS_DIR = Path(__file__).resolve().parents[1] / "prompts"


def load_prompt(filename: str, vars: dict = {}) -> str:
    template = (PROMPTS_DIR / filename).read_text(encoding="utf-8")
    for key, value in vars.items():
        template = template.replace(f"{{{{{key}}}}}", value)
    return template
